### ISSUE TITLE
Linear Regression Model

### ISSUE THREAD

**Issue by Maxime-Cy** [Maxime-Cy/Applied-python#6]
import pandas as pd
import os
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score
import matplotlib.pyplot as plt

# File path and column names
pathfile = "https://github.com/user-attachments/files/20428088/Data.HSS.csv"
columns = ["TV", "Radio", "Social Media", "Influencer", "Sales"]

# Load dataset from CSV if it exists
if os.path.exists(pathfile):
    try:
        df = pd.read_csv(pathfile, header=0)
    except pd.errors.EmptyDataError:
        df = pd.DataFrame(columns=columns)
else:
    df = pd.DataFrame()

# Drop any rows with missing values
df = df.dropna()

# Define features and target variable for the model
X = df[["Radio"]]
y = df["Sales"]

# Split the data into training and testing sets (80% train, 20% test)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Initialize and train the linear regression model
model = LinearRegression()
model.fit(X_train, y_train)

# Predict sales using the test set
y_pred = model.predict(X_test)

# Evaluate the model performance
mse = mean_squared_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

# Output model parameters and performance metrics
print("\nModel Coefficients:")
print("Slope:", model.coef_[0])
print("Intercept:", model.intercept_)
print("\nModel Performance:")
print("Mean Squared Error (MSE)", mse)
print("R-squared:", r2)

# Visualize the results with a scatterplot and the regression line
plt.scatter(X_test, y_test, color='blue', label='actual points')
plt.plot(X_test, y_pred, color='red', label='predicted points')
plt.title('Actual vs Predicted Sales')
plt.xlabel('Radio promotion budget')
plt.ylabel('Sales')
plt.legend()
plt.tight_layout()
plt.show()
